rules/aip0203/field_behavior_required.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package aip0203
+
+import (
+	"fmt"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var requiredOneOfFieldBehavior = stringset.New(
+	"OPTIONAL", "REQUIRED", "OUTPUT_ONLY",
```

Comment:
Should this include `IMMUTABLE`?

Reply:
Is immutable enough by itself? I figured that we need immutable + (optional | required) to fully define the field. 

there's also discussion about immutable + output_only being valid, with immutable just stating it doesn't change for the lifecycle of the request.

Reply:
IMO, `IMMUTABLE` is enough. It has sufficient semantic meaning on its own but maybe check with others too?

Reply:
is an immutable field required? 

for a third party, we could maybe loop in @jskeet or @noahdietz?

The result would be an AIP change most likely.

Reply:
I wouldn't say that Immutable is sufficient to communicate if the field is required or not. An Immutable field could be optional, and the service default is used if unset, which then cannot be changed.

Immutable only conveys the mutability of a field post-resource creation. It does not convey what constitutes a valid request.

Reply:
The md suggests that it checks if the field_behavior has a valid value not if it is sufficient to determine if it is required. IMO, `IMMUTABLE` is a valid value.

Reply:
ack. would you prefer a separate rule to force adding one of REQUIRED / OPTIONAL?

This is what the corresponding AIP change is proposing (which, BTW, needs reviews as well if someone is able to spare some time: https://github.com/aip-dev/google.aip.dev/pull/1100)

Reply:
I don't think immutable is sufficient information for a client. 

For example, if it's `IMMUTABLE`, is the field required or optional? IMO a client can't generate itself unless it know that answer.

Reply:
So I stand by what I said, but the linter seem to imply that OPTIONAL is mutually exclusive.

I've written an issue but for this and I'll round back (https://github.com/aip-dev/google.aip.dev/issues/1119). For now I'll accept that IMMUTABLE by itself is a valid annotation, because somehow it implies required.

Reply:
> IMMUTABLE by itself is a valid annotation, because somehow it implies required.

That's news to me 🙃

Reply:
> I don't think immutable is sufficient information for a client.
> 
> For example, if it's `IMMUTABLE`, is the field required or optional? IMO a client can't generate itself unless it know that answer.

Ack. Do we say anywhere in an AIP that use of `IMMUTABLE` **must** be accompanied by another FieldBehavior value?

Reply:
no, although it is noted that *optional* can only be used if no other annotation is used (which, IMO, is not great behavior).

I've filed https://github.com/aip-dev/google.aip.dev/pull/1120 to fix that. I'll follow-up with another CL after this one once the referenced PR is merged.